pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package knativeserving
+
+import (
+	"context"
+	"knative.dev/serving-operator/pkg/reconciler/knativeserving/common"
+
+	mf "github.com/jcrossley3/manifestival"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"knative.dev/pkg/controller"
+	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+	listers "knative.dev/serving-operator/pkg/client/listers/serving/v1alpha1"
+	"knative.dev/serving-operator/pkg/reconciler/newreconciler"
+	"knative.dev/serving-operator/version"
+)
+
+var (
+	// Platform-specific behavior to affect the installation
+	platforms common.Platforms
+)
+
+// Reconciler implements controller.Reconciler for Knativeserving resources.
+type Reconciler struct {
+	*newreconciler.Base
+	// Listers index properties about resources
+	knativeServingLister          listers.KnativeServingLister
+	deploymentLister              appsv1listers.DeploymentLister
+	serviceLister                 corev1listers.ServiceLister
+	config                        mf.Manifest
+
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Knativeserving resource
+// with the current status of the resource.
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
```

Comment:
Why are we creating a full blown copy of this? Makes diffing this against the old reconciler impossible.